qcodes/instrument_drivers/american_magnetics/AMI430_visa.py

```diff
@@ -0,0 +1,1110 @@
+            "state",
+            label="Switch Heater On",
+            get_cmd=self.check_state,
+            set_cmd=lambda x: (self.on() if x else self.off()),
+            vals=Bool(),
+        )
+        self.add_parameter(
+            "in_persistent_mode",
+            label="Persistent Mode",
+            get_cmd="PERS?",
+            val_mapping={True: 1, False: 0},
+        )
+
+        # Configuration Parameters
+        self.add_parameter(
+            "current",
+            label="Switch Heater Current",
+            unit="mA",
+            get_cmd="PS:CURR?",
+            get_parser=float,
+            set_cmd="CONF:PS:CURR {}",
+            vals=Numbers(0, 125),
+        )
+        self.add_parameter(
+            "heat_time",
+            label="Heating Time",
+            unit="s",
+            get_cmd="PS:HTIME?",
+            get_parser=int,
+            set_cmd="CONF:PS:HTIME {}",
+            vals=Ints(5, 120),
+        )
+        self.add_parameter(
+            "cool_time",
+            label="Cooling Time",
+            unit="s",
+            get_cmd="PS:CTIME?",
+            get_parser=int,
+            set_cmd="CONF:PS:CTIME {}",
+            vals=Ints(5, 3600),
+        )
+
+    def disable(self) -> None:
+        """Turn measurement off"""
+        self.write("CONF:PS 0")
+        self._enabled = False
+
+    def enable(self) -> None:
+        """Turn measurement on"""
+        self.write("CONF:PS 1")
+        self._enabled = True
+
+    def check_enabled(self) -> bool:
+        return bool(int(self.ask("PS:INST?").strip()))
+
+    @_Decorators.check_enabled
+    def on(self) -> None:
+        self.write("PS 1")
+        while self._parent.ramping_state() == "heating switch":
+            self._parent._sleep(0.5)
+
+    @_Decorators.check_enabled
+    def off(self) -> None:
+        self.write("PS 0")
+        while self._parent.ramping_state() == "cooling switch":
+            self._parent._sleep(0.5)
+
+    @_Decorators.check_enabled
+    def check_state(self) -> bool:
+        return bool(int(self.ask("PS?").strip()))
+
+
+class AMI430(VisaInstrument):
+    """
+    Driver for the American Magnetics Model 430 magnet power supply programmer.
+
+    This class controls a single magnet power supply. In order to use two or
+    three magnets simultaneously to set field vectors, first instantiate the
+    individual magnets using this class and then pass them as arguments to
+    either the AMI430_2D or AMI430_3D virtual instrument classes.
+
+    Args:
+        name: a name for the instrument
+        address: IP address of the power supply programmer
+        current_ramp_limit: A current ramp limit, in units of A/s
+    """
+
+    _SHORT_UNITS = {"seconds": "s", "minutes": "min", "tesla": "T", "kilogauss": "kG"}
+    _DEFAULT_CURRENT_RAMP_LIMIT = 0.06  # [A/s]
+
+    def __init__(
+        self,
+        name: str,
+        address: str,
+        reset: bool = False,
+        terminator: str = "\r\n",
```

Comment:
noticed that the terminator used in the tests with the sim is actually ``\n``. could be relevant or not, dont know

Reply:
I was wondering about that too. Looking at the log message in the traceback the read terminator is clearly \r\n

Reply:
ValueError: ("could not convert string to float: '0.001666667,1.0\\r\\n'", 'setting magnet_x_field to 0.05')